src/support/import-supervisor.js

```diff
@@ -21,54 +37,181 @@ function ImportSupervisor(services) {
   }
 
   validateServices();
-  // eslint-disable-next-line no-unused-vars
-  const { log, sqsClient, s3Client } = services;
+  const {
+    dataAccess, sqs, s3: { s3Client, GetObjectCommand, getSignedUrl }, log, env,
+  } = services;
+  const {
+    IMPORT_QUEUE_URL_PREFIX,
+    IMPORT_QUEUES, // Comma separated list of import queues
+    IMPORT_S3_BUCKET,
+  } = env;
+  const IMPORT_RESULT_ARCHIVE_NAME = 'import-result.zip';
+
+  /**
+   * Get an available import queue name that is not currently in use. Throws an error if no queue
+   * is currently available.
+   */
+  async function getAvailableImportQueue() {
+    const runningImportJobs = await dataAccess.getImportJobsByStatus(IMPORT_JOB_STATUS.RUNNING);
+    const importQueues = IMPORT_QUEUES.split(',');
+
+    // Find an import queue that is not in use
+    for (const queue of importQueues) {
+      if (!runningImportJobs.includes(queue)) {
+        return queue;
+      }
+    }
+    throw new ErrorWithStatusCode('Service Unavailable: No import queue available', 503);
+  }
+
+  function determineBaseURL(urls) {
+    // Initially, we will just use the domain of the first URL
+    const url = new URL(urls[0]);
+    return `${url.protocol}//${url.hostname}`;
+  }
+
+  /**
+   * Create a new import job by claiming one of the free import queues, persisting the import job
+   * metadata, and setting the job status to 'RUNNING'.
+   * @param {Array<string>} urls - The list of URLs to import.
+   * @param {string} importQueueId - Name of the queue to use for this import job.
+   * @param {string} apiKey - API key used to authenticate the import job request.
+   * @param {object} options - Client provided options for the import job.
+   * @returns {Promise<ImportJob>}
+   */
+  async function createNewImportJob(urls, importQueueId, apiKey, options) {
+    const newJob = {
+      id: crypto.randomUUID(),
+      baseURL: determineBaseURL(urls),
+      importQueueId,
+      apiKey,
+      options,
+      status: IMPORT_JOB_STATUS.RUNNING,
+    };
+    return dataAccess.createNewImportJob(newJob);
+  }
+
+  /**
+   * Persist the URLs to import in the data layer, each as an import URL record.
+   */
+  async function persistUrls(jobId, urls) {
+    // - Generate a urlId guid for this single URL
+    // - Set status to 'pending'
+    const urlRecords = [];
+    for (const url of urls) {
+      const urlRecord = {
+        id: crypto.randomUUID(),
+        jobId,
+        url,
+        status: 'PENDING',
+      };
+      // eslint-disable-next-line no-await-in-loop
```

Comment:
is this done for stability considerations? otherwise suggest `Promise.all`

Reply:
My thinking was that there wasn't anything to be gained (performance wise) by not waiting in between requests, since the import itself will be performed in serial. I would be fine to go with `Promise.all` if preferred, but I do find this approach quite readable.